src/axom/quest/readers/C2CReader.cpp

```diff
@@ -162,598 +166,47 @@ static void appendPoints(mint::UnstructuredMesh<mint::SINGLE_SHAPE>* mesh,
-      for(int i = 1; i <= k; i++)
-      {
-        auto bin = axom::utilities::binomialCoefficient(k, i);
-        v[0] = v[0] - bin * wders[i] * CK[k - i][0];
-        v[1] = v[1] - bin * wders[i] * CK[k - i][1];
-      }
-      CK[k][0] = v[0] / wders[0];
-      CK[k][1] = v[1] / wders[0];
-    }
-  }
-
-  /*!
-   * \brief Evaluates the B-spline curvature at parameter value \a u
-   * 
-   * \param u The parameter value.
-   *
-   * \return The curvature value at u.
-   */
-  double curvature(double u) const
-  {
-    // Evaluate 1st and 2nd derivatives at u.
-    PointType derivs[3];
-    derivativesAt(u, 2, derivs);
-    const PointType& D1 = derivs[1];
-    const PointType& D2 = derivs[2];
-
-    double xp = D1.data()[0];   // x'
-    double xpp = D2.data()[0];  // x''
-
-    double yp = D1.data()[1];   // y'
-    double ypp = D2.data()[1];  // y''
-
-    // This is signed curvature as formulated at:
-    // https://en.wikipedia.org/wiki/Curvature#Curvature_of_a_graph
-    // k = (x'y'' - y'x'') / pow(x'x' + y'y', 3./2.)
-    double xp2_plus_yp2 = xp * xp + yp * yp;
-    return (xp * ypp - yp * xpp) / pow(xp2_plus_yp2, 3. / 2.);
-  }
-
-  /*!
-   * \brief Evaluates the B-spline curvature derivatives (up to 2nd) at
-   *        parameter value \a u
-   * 
-   * \param[in] u The parameter value.
-   * \param[in] d The number of derivatives to compute (1=1st deriv, 2=1st & 2nd derivs)
-   * \param[out] ders An array that will contain the curvature derivatives.
-   */
-  void curvatureDerivatives(double u, int d, double* ders) const
-  {
-    // Evaluate 1st, 2nd, 3rd curve derivatives at u.
-    PointType derivs[4];
-    derivativesAt(u, 3, derivs);
-    const PointType& D1 = derivs[1];
-    const PointType& D2 = derivs[2];
-    const PointType& D3 = derivs[3];
-
-    double xp = D1.data()[0];    // x'
-    double xpp = D2.data()[0];   // x''
-    double xppp = D3.data()[0];  // x'''
-
-    double yp = D1.data()[1];    // y'
-    double ypp = D2.data()[1];   // y''
-    double yppp = D3.data()[1];  // y'''
-
-    // 1st derivative of curvature.
-    double xp2_plus_yp2 = xp * xp + yp * yp;
-    double A = -3. * (xp * ypp - yp * xpp) * 2 * (xp * xpp + yp * ypp);
-    double B = 2. * pow(xp2_plus_yp2, 5. / 2.);
-    double C = xp * yppp - yp * xppp;
-    double D = pow(xp2_plus_yp2, 3. / 2.);
-    ders[0] = A / B + C / D;
-
-    if(d >= 2)
-    {
-      // 2nd derivative of curvature.
-      double E = 15. * (-yp * xpp + xp * ypp) *
-        pow(2. * xp * xpp + 2. * yp * ypp, 2.) /
-        (4. * pow(xp2_plus_yp2, 7. / 2.));
-      double F = 3. * (2. * xp * xpp + 2. * yp * ypp) *
-        (-yp * xppp + xp * yppp) / pow(xp2_plus_yp2, 5. / 2.);
-      double G = 3. * (-yp * xpp + xp * ypp) *
-        (2. * (xpp * xpp) + 2. * (ypp * ypp) + 2. * xp * xppp + 2. * yp * yppp) /
-        (2. * pow(xp2_plus_yp2, 5. / 2.));
-      double H = (-ypp * xppp + xpp * yppp) / pow(xp2_plus_yp2, 3. / 2.);
-
-      ders[1] = E - F - G + H;
-    }
-  }
```

Comment:
@BradWhitlock -- curvature is the only functionality that is being removed by this PR. It wasn't being used in the code in this file.  Should we add it to the `NURBSCurve` and `BezierCurve` classes?

Reply:
I wrote the curvature functions when I was looking for a sample non-uniformly and ended up using a different algorithm. Moving these to the curve classes sounds like a good idea. That's where they'd be if those classes existed at the time.

Reply:
#1530